src/lib/operators/insert.cpp

```diff
@@ -105,13 +105,22 @@ std::shared_ptr<const Table> Insert::_on_execute(std::shared_ptr<TransactionCont
     while (remaining_rows > 0) {
       auto target_chunk_id = ChunkID{_target_table->chunk_count() - 1};
       auto target_chunk = _target_table->get_chunk(target_chunk_id);
+      auto mvcc_data = target_chunk->mvcc_data();
 
-      // If the last Chunk of the target Table is either immutable or full, append a new mutable Chunk
+      // If the last chunk of the target table is either immutable or full, append a new mutable chunk.
       if (!target_chunk->is_mutable() || target_chunk->size() == _target_table->target_chunk_size()) {
+        // Allow the chunk to be finalized since no new Insert operators will write to it. Then try to finalize it. If
+        // all Insert operators writing to the chunk are already finished, this make the chunk immutable. Otherwise the
```

Comment:
better?
```c++
// Allow the chunk to be finalized since no new Insert operators will write to it. Then try to finalize it
// because the chunk reached its target size. If there are no other pending Insert operators inserting into this
// chunk, `try_finalize()` makes the chunk immutable. Otherwise, the pending Insert operators will do this once
// they are done.
```

Reply:
Merge the first two sentences? "Allow the chunk to be finalized since it has reached target size & no more Writers."? 
The third sentence "If there are no ..." again sounds like the first sentence.

Reply:
But there _is_ a difference between the first and the third sentence: no new incoming Insert operators will try to insert into the chunk vs. Insert operators that try to insert are finished.

Reply:
reformulated to:
```c++
// Allow the chunk to be finalized as it has reached its target size and no incoming Insert operators will try
// to write to it. If there are no other Insert operators still inserting into this chunk, `try_finalize()`
// makes the chunk immutable. Otherwise, the pending Insert operators will do this once they are done.
```

Reply:
(is the comment already committed? than feel free to resolve)